### PR TITLE
Fixes #15665: ensure parent state url matches before 404.

### PR DESCRIPTION
**Pull request by waldenraines**
Adding an additional check to ensure that a potential 404 is valid.
This issue can be encountered if the state's url doesn't match its name.

http://projects.theforeman.org/issues/15665

### CODE REVIEW

**Comment by ehelms** [comment]
I wonder if we should write some unit tests for this? Might not be easy in it's current form, and would potentially need to re-factor to a service. However, given the trouble we've had with this, it might be worth it for regressions? Open to your thoughts, but thought I would propose the idea.

**Comment by waldenraines** [comment]
@ehelms updated.  There are a lot of changes so review carefully 😉 

Also, I want to keep these as two separate commits so we can include just the fix downstream if we need to.

File: app/assets/javascripts/bastion/components/current-tasks.directive.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ angular.module('Bastion.components').directive('currentTasks',
                 });
             }],
             link: function (scope) {
-                Task.registerSearch({ 'active_only': true, 'type': 'user', 'user_id': CurrentUser.id}, scope.updateTasks);
+                Task.registerSearch({ 'active_only': true, 'type': 'user', 'user_id': authorizationService.id}, scope.updateTasks);
```
Comment (ehelms) [inline app/assets/javascripts/bastion/components/current-tasks.directive.js]: Should this be `authorizationService.getCurrentUser().id` ?

File: app/views/bastion/layouts/application.html.erb [hunk 2]
```diff
@@ -1,3 +1,7 @@
+<% content_for(:head) do %>
+    <base href="/bastion"/>
```
Comment (ehelms) [inline app/views/bastion/layouts/application.html.erb]: What does the addition of this do?

File: app/assets/javascripts/bastion/components/current-tasks.directive.js (hunk 1 above)
Comment (waldenraines) [inline app/assets/javascripts/bastion/components/current-tasks.directive.js]: Yes, good catch, will fix.

File: app/views/bastion/layouts/application.html.erb (hunk 2 above)
Comment (waldenraines) [inline app/views/bastion/layouts/application.html.erb]: https://github.com/angular-ui/ui-router/wiki/Frequently-Asked-Questions#issue-my-assets-and-templates-are-not-loading

Without it I was getting a javascript error when attempting to go to any new katello page.

**Comment by waldenraines** [comment]
@ehelms updated.

**Comment by ehelms** [comment]
I noticed in testing that this brings back the infinite loop for non-existent states. If go to `/activation_keys/145` for example, I never got to a 404 page.

**Comment by waldenraines** [comment]
> I noticed in testing that this brings back the infinite loop for non-existent states. If go to /activation_keys/145 for example, I never got to a 404 page.

Nice catch, I have fixed this in the latest push.

**Comment by ehelms** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by ehelms** [comment]
I am seeing one minor weirdness where if I go to `/lifecycle_environments/100` (one that does not exist) it takes me to a blank details page.

**Comment by waldenraines** [comment]
> I am seeing one minor weirdness where if I go to /lifecycle_environments/100 (one that does not exist) it takes me to a blank details page.

That was the case before my change, can we tackle that as a separate issue?

**Comment by ehelms** [comment]
@waldenraines Sure.

ACK

**Comment by waldenraines** [comment]
http://projects.theforeman.org/issues/16246

**Pull request merged**